src/transaction/transaction.rs

```diff
@@ -450,8 +498,8 @@ impl TwoPhaseCommitter {
             .await
     }
 
-    fn rollback(&mut self) -> impl Future<Output = Result<()>> + 'static {
-        let mutations = mem::take(&mut self.mutations);
+    fn rollback(&self) -> impl Future<Output = Result<()>> + 'static {
```

Comment:
Could we take `self` by value and avoid the clone?

Reply:
I don't think so. If the rollback request failed, we need to keep the data.

Reply:
I don't think we need to keep the data on failure. Rollback failure rarely happens so it's ok to just leave these garbage on TiKV.

Reply:
My mistake, it's inside `TwoPhaseCommiter` which is not reused.

Reply:
I think we might want to let the user retry a rollback, or at least we might want to let them retry a commit, and I think the commit and rollback APIs should be similar.

Reply:
So how about this:
Dropping `Trarnsaction` before any rollback or commit => panic,
Commit or rollback success => cannot do anything further
Commit or rollback failure => can only retry, will not panic on drop

Reply:
This sounds good. In addition, I think `TwoPhaseCommitter` can consume `self` on rollback or commit?

Reply:
:+1:

Reply:
Yeah. They already consumed `self`.